### DISCUSSION TITLE
Cannot link to a non running container while trying the getting started guide

### DISCUSSION THREAD

**Discussion by gerardnico** [dokku/dokku#6917] [Q&A]
Hallo,

I try to push my first app against Dokku following the [application-deployment](https://dokku.com/docs/deployment/application-deployment/) and I get a fail deployment with the following message: `Cannot link to a non running container`

This is pretty scary and I don't understand why a container could not run if another container does not run.
It should just be unreachable? No? Did I miss something?

Thanks

Logs:
```bash
git push dokku main
```

Output:
```
Enumerating objects: 579, done.
Counting objects: 100% (579/579), done.
Delta compression using up to 16 threads
Compressing objects: 100% (283/283), done.
Writing objects: 100% (579/579), 142.52 KiB | 17.81 MiB/s, done.
Total 579 (delta 265), reused 579 (delta 265), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (265/265), done.
-----> Set main as deploy-branch
-----> Cleaning up...
-----> Building getting-started from herokuish
-----> Setting config vars
       CURL_CONNECT_TIMEOUT:  90
-----> Setting config vars
       CURL_TIMEOUT:  600
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
remote: Error response from daemon: Cannot link to a non running container: /dokku.postgres.railsdatabase AS /zealous_elion/dokku-postgres-railsdatabase
remote: Error: failed to start containers: 8f302668f992c7450451de7e0f4ee8f0f6a547e90422ce1f2f0b1cc5dcd3b232
remote: You cannot attach to a stopped container, start it first
remote:  !     Failure during app build
remote:  !     Removing invalid image tag dokku/getting-started:latest
remote:  !     App build failed
To ssh://localhost:3022/getting-started
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'ssh://localhost:3022/getting-started'
```

**Comment by josegonzalez** [comment]
For some reason, your postgres container isn't running. If you try to link a container to that postgres container, Docker won't start the new container. Unfortunately, we can't change how that works in Docker.

You'll want to debug whats going on with the postgres container - maybe it didn't get created properly, or maybe it crashed. You can check the `postgres:info` and `posgres:logs` commands to investigate further.

**Reply by gerardnico** [reply]
Thanks.

You should not use it if this how it works. The tuto should just give a container hostname.

An app may work well without a third service.

Link is legacy. I have never used it.
https://docs.docker.com/network/links/

Thanks for the answer. All the best
